### PR TITLE
CREATE EXTENSION always requires superuser.

### PR DESCRIPTION
**Pull request by minusf**
For non-superuser cases skip the command enirely if the extension is
already installed.

### CODE REVIEW

File: django/contrib/postgres/operations.py [hunk 1]
```diff
@@ -21,7 +21,14 @@ def database_forwards(self, app_label, schema_editor, from_state, to_state):
             not router.allow_migrate(schema_editor.connection.alias, app_label)
         ):
             return
-        schema_editor.execute("CREATE EXTENSION IF NOT EXISTS %s" % schema_editor.quote_name(self.name))
+
+        # CREATE EXTENSION always requires superuser. For non-superuser cases
+        # skip the command enirely if the extension is already installed.
```
Comment (NyanKiyoshi) [inline django/contrib/postgres/operations.py:26]:
```suggestion
        # skip the command entirely if the extension is already installed.
```

**Comment by minusf** [comment]
```
Traceback (most recent call last):
  File "/home/jenkins/workspace/pull-requests-bionic/database/postgres/label/bionic-pr/python/python3.8/tests/postgres_tests/test_operations.py", line 183, in test_allow_migrate
    self.assertIn('CREATE EXTENSION', captured_queries[0]['sql'])
AssertionError: 'CREATE EXTENSION' not found in "SELECT * FROM pg_extension WHERE extname = 'uuid-ossp'"
```

it seems that this [test](https://github.com/django/django/blob/master/tests/postgres_tests/test_operations.py#L175) was always a NOP because of `IF NOT EXISTS`. At this point in the test run the extension is already there, this is what i see while in `pdb`:

```
test_extension_checks=# \dx
                                     List of installed extensions
    Name    | Version |   Schema   |                            Description
------------+---------+------------+-------------------------------------------------------------------
 bloom      | 1.0     | public     | bloom access method - signature file based index
 btree_gin  | 1.3     | public     | support for indexing common datatypes in GIN
 btree_gist | 1.5     | public     | support for indexing common datatypes in GiST
 citext     | 1.6     | public     | data type for case-insensitive character strings
 hstore     | 1.6     | public     | data type for storing sets of (key, value) pairs
 pg_trgm    | 1.4     | public     | text similarity measurement and index searching based on trigrams
 pgcrypto   | 1.3     | public     | cryptographic functions
 plpgsql    | 1.0     | pg_catalog | PL/pgSQL procedural language
 unaccent   | 1.1     | public     | text search dictionary that removes accents
 uuid-ossp  | 1.1     | public     | generate universally unique identifiers (UUIDs)
(10 rows)
```

without going through the whole testsuite and trying to figure out what extensions is installed when, a rather hackish solution is to install a hitherto unused module and bump the index to 1. 🙈

**Comment by minusf** [comment]
oh git...  pulling in `505f45d` was unintended, sorry.

**Comment by NyanKiyoshi** [comment]
@minusf `git rebase -i upstream/master`, then remove the commit (line)

**Comment by minusf** [comment]
sorry the rebase was not working for me, that commit was not showing up in the list at all.  i will make a new PR.

**Comment by minusf** [comment]
https://github.com/django/django/pull/12952

**Pull request closed without merging**